src/snowflake/connector/aio/auth/_keypair.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+from __future__ import annotations
+
+from logging import getLogger
+from typing import Any
+
+from cryptography.hazmat.primitives.asymmetric.rsa import RSAPrivateKey
+
+from ...auth.keypair import AuthByKeyPair as AuthByKeyPairSync
+from ._by_plugin import AuthByPlugin
+
+logger = getLogger(__name__)
+
+
+class AuthByKeyPair(AuthByPlugin, AuthByKeyPairSync):
+    def __init__(
+        self,
+        private_key: bytes | RSAPrivateKey,
+        lifetime_in_seconds: int = AuthByKeyPairSync.LIFETIME,
+        **kwargs,
+    ) -> None:
+        AuthByKeyPairSync.__init__(self, private_key, lifetime_in_seconds, **kwargs)
+
+    async def reset_secrets(self) -> None:
+        AuthByKeyPairSync.reset_secrets(self)
+
+    async def prepare(self, **kwargs: Any) -> None:
+        AuthByKeyPairSync.prepare(self, **kwargs)
+
+    async def reauthenticate(self, **kwargs: Any) -> dict[str, bool]:
+        return AuthByKeyPairSync.reauthenticate(self, **kwargs)
+
+    async def update_body(self, body: dict[Any, Any]) -> None:
```

Comment:
Same questions here, do we need to make these function async?

Reply:
the async def is part of the protocol for our public auth plug in class.